language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -119,7 +123,7 @@ instance ExMemoryUsage () where
   memoryUsage _ = 0 -- TODO or 1?
 
 instance ExMemoryUsage Integer where
-  memoryUsage _ = 2 -- TODO
+  memoryUsage i = ExMemory (smallInteger (integerLog2# (abs i) `quotInt#` (integerToInt 60))) -- assume 60bit size
```

Comment:
Where did you find said `integerLog2`?

Reply:
I was looking [here](https://hackage.haskell.org/package/arithmoi-0.4.0.3/docs/src/Math-NumberTheory-Logarithms.html#integerLog2).

Reply:
If you look at the implementation, it does exactly the same :-)

```
integerLog2 :: Integer -> Int
integerLog2 n
  | n < 1       = error "Math.NumberTheory.Logarithms.integerLog2: argument must be positive"
  | otherwise   = I# (integerLog2# n)
```

Reply:
I did look at the implementation, which is why I suggested using some other function than `integerLog2#`, which seems to rely on the `n < 1` invariant, which it doesn't seem to check. You could introduce your own wrapper around `integerLog2#` doing the necessary invariant check.

Reply:
Yeah, I added said check inline. Dunno if that's the right way to go.